File: backend/api/repositories/user_repository.py
"""User repository for database operations"""
from typing import Optional, Dict
from .base import BaseRepository


class UserRepository(BaseRepository):
    """Repository for user operations"""
    
    def __init__(self, db):
        super().__init__(db, "users")
    
    async def find_by_email(self, email: str) -> Optional[Dict]:
        """Find user by email"""
        return await self.find_one({"email": email})
    
    async def get_users_by_role(self, role: str) -> list:
        """Get all users with specific role"""
        return await self.find_many({"role": role})

File: nextjs-app/src/lib/repositories/user.repository.ts
import { BaseRepository } from './base.repository';

interface User {
  id?: string;
  email: string;
  password: string;
  name: string;
  phone?: string;
  role?: string;
  created_at?: Date;
  updated_at?: Date;
}

export class UserRepository extends BaseRepository<User> {
  constructor() {
    super('users');
  }

  async findByEmail(email: string): Promise<User | null> {
    return this.findOne({ email });
  }

  async findByRole(role: string): Promise<User[]> {
    return this.findAll({ role });
  }
}
